### PR TITLE
CategoryView: use filter_func instead of destroy

### PR DESCRIPTION
**Pull request by danrabbit**
Because creating and destroying widgets is wasteful

### CODE REVIEW

File: src/Views/CategoryView.vala [hunk 1]
```diff
@@ -216,16 +222,18 @@ public class Slingshot.Widgets.CategoryView : Gtk.EventBox {
...
-            child.destroy ();
-        }
-
-        foreach (Backend.App app in view.app_system.apps[category]) {
-            listbox.add (new AppListRow (app.desktop_id, app.desktop_path));
+    [CCode (instance_pos = -1)]
+    private bool filter_function (AppListRow row) {
+        unowned CategoryRow category_row = (CategoryRow) category_switcher.get_selected_row ();
+        if (category_row != null) {
+            foreach (Backend.App app in view.app_system.apps[category_row.cat_name]) {
```
Comment (jeremypw) [inline src/Views/CategoryView.vala:229]: I think this can be `unowned Backend.App`?  It is in a similar iteration elsewhere.

**Review by jeremypw — CHANGES_REQUESTED** [review]
Just a tiny comment otherwise works as expected.

File: src/Views/CategoryView.vala (hunk 1 above)
Comment (danrabbit) [reply src/Views/CategoryView.vala:229]: Looks like no, I get `Invalid assignment from owned expression to unowned variable`

Comment (jeremypw) [reply src/Views/CategoryView.vala:229]: Sorry, in the other case the loop was iterating a list returned from a function `get_apps_by_name` but you are accessing the Gee.ArrayList directly.

**Review by jeremypw — APPROVED** [review]

**Pull request merged**